searching/floyd_cycle_detection_algorithm.c

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file floyd_cycle_detection_algorithm.c
+ * @brief Implementation of [Floyd's Cycle
+ * Detection](https://en.wikipedia.org/wiki/Cycle_detection) algorithm
+ * @details
+ * Given an array of integers containing 'n + 1' integers, where each
+ * integer is in the range [1, n] inclusive. If there is only one duplicate
+ * number in the input array, this algorithm returns the duplicate number in
+ * O(1) space and the time complexity is less than O(n^2) without modifying the
+ * original array, otherwise, it returns -1.
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+
+/**
+ * @brief The main function implements search algorithm
+ * @tparam T type of array
+ * @param in_arr the input array
+ * @param n size of array
+ * @returns the duplicate number
+ */
+int duplicateNumber(const int *in_arr, size_t n)
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. **Requires adding the `inttypes.h` library.** Check other parts of the code ([reference](https://en.cppreference.com/w/c/types/integer)). 🙂